faucet/valve.py

```diff
@@ -1361,6 +1366,23 @@ def get_config_dict(self):
             'acls': acls_dict,
             }
 
+    def flow_timeout(self, table_id, match):
+        eth_src = None
+        match_oxm_fields = match.to_jsondict()['OFPMatch']['oxm_fields']
+        if table_id == self.dp.eth_src_table:
```

Comment:
That does mean we have potentially 5 minutes of flooding right? It does seem plausible that on a large network, there would just be a constant number of hosts that you can expect to have all their traffic flooded at any given time.

When we could avoid this just by having a rule that catches when eth_dst rules time out and reinstalls them or removes the src matches to force them to be relearnt. Or we could have them never time out, and just have them be cleared every expire period, (if there is a sensible way to have the cache clear them, I am not sure off the top of my head what that would look like).